urllib3/connection.py

```diff
@@ -162,6 +162,34 @@ def connect(self):
         conn = self._new_conn()
         self._prepare_conn(conn)
 
+    def request_chunked(self, method, url, body=None, headers={}):
+        """
+        Alternative to the common request method, which sends the
+        body with chunked encoding and not as one block
+        """
+        header_names = set(k.lower() for k in headers)
+        self.putrequest(method, url, skip_accept_encoding=True)
+        for header, value in headers.items():
+            self.putheader(header, value)
+        if 'transfer-encoding' not in header_names:
+            self.putheader('Transfer-Encoding', 'chunked')
+        self.endheaders()
+
+        if body is not None:
+            if isinstance(body, six.string_types):
+                body = [body]
+            for chunk in body:
+                if not chunk:
+                    continue
+                len_str = hex(len(chunk))[2:]
+                self.send(len_str.encode('utf-8'))
+                self.send(b'\r\n')
+                self.send(chunk.encode('utf-8'))
+                self.send(b'\r\n')
```

Comment:
In terms of unicode, please have a look at https://github.com/python-mirror/python/blob/2.7/Lib/httplib.py#L814 this is the part where httplib is sending it's body. No check on anything. Feed in a unicode string, and it tries to send it as a file etc.

Now of course there is no point in repeating the mistakes from httplib. Still, httplib relies on the higher level API to present sane input to `request`. The same should be expected for `request_chunked`, as it's supposed to serve as a drop in.